bundle-workflow/lib/assemble.py

```diff
@@ -0,0 +1,47 @@
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  The OpenSearch Contributors require contributions made to
+#  this file be licensed under the Apache-2.0 license or a
+#  compatible open source license.
+
+import os
+import sys
+import tempfile
+from src.bundle_manifest import BundleManifest
+from src.build_output import BuildOutput
+from src.bundle import Bundle
+
+if (len(sys.argv) < 2):
+    print("Build an OpenSearch Bundle")
+    print("usage: assemble.sh /path/to/build_manifest")
+    exit(1)
+
+with tempfile.TemporaryDirectory() as work_dir:
+    manifest = BundleManifest.from_file(sys.argv[1])
+    build = manifest.build
+    output = BuildOutput(os.path.join(os.getcwd(), 'bundles'))
+
+    print(f'Bundling {build.name} ({output.arch}) into {output.dest} ...')
+
+    os.chdir(work_dir)
+
+    bundle = Bundle(manifest.min_bundle)
+
+    for plugin in manifest.plugins:
+        print(f'=== Installing {plugin.name} ...')
+        plugin.download(work_dir)
+        bundle.install_plugin(plugin)
+
+    print(f'Installed plugins:')
+    print(bundle.plugins)
+
+    filename = "-".join([build.name.lower(), build.version, 'linux', output.arch]) + '.tar.gz'
+    build.dict['location'] = os.path.join(output.dest, filename)
```

Comment:
Let’s add a todo to fix this? Seems equally weird to the old version.

Reply:
+1 will see how this looks with @camerski 's refactoring in #165.